.github/workflows/beta.yml

```
@@ -0,0 +1,20 @@
+name: Beta branch
+
+on:
+  push:
+    branches:
+      - 'beta'
+
+jobs:
+  unit-test:
+    uses: ./.github/workflows/job-test.yml
+
+  check-new-version:
+    needs: unit-test
+    uses: ./.github/workflows/job-check-new-version.yml
+
+  publish:
+    needs: check-new-version
+    if: ${{ needs.check-new-version.outputs.isNewVersion == 'true' }}
+    uses: ./.github/workflows/job-publish.yml
+    secrets: inherit
```

Two suggestions:

- beta and main workflows can be merged? I don't see a difference
- I don't think you need to manage the check for newVersion manually. You can just rely on the status code of running the script and if it's a success (zero), continue. Otherwise, abort.

I think these changes could make the scripts easier to follow.


```suggestion
name: publish

on:
  push:
    branches:
      - 'beta'
      - 'main'

jobs:
  unit-test:
    uses: ./.github/workflows/job-test.yml

  check-new-version:
    id: checkNewVersion
    needs: unit-test
    run: sh ./.github/workflows/check-new-version.sh

  publish:
    needs: check-new-version
    if: steps.checkNewVersion.outputs.status == 'failure' 
    uses: ./.github/workflows/job-publish.yml
    secrets: inherit
```

---

Thanks for the comment @kabaros.
- Totally agree on the first point. Those workflows were different at the beginning; they ended up with the same structure and I forgot to remove one of them.
- About the second point, it would be easier just to check for a failure as you suggested. The only reason why I preferred to check the output was to make the step not fail when the version has not change. The overall result would be the same (the version won't be published), but the "check-new-version" step will be colored in red, which could be considered a false positive because it is not an error, it is a valid. What do you think? I am not a pro with Github actions, maybe there is another way to deal with it